### PR TITLE
[usbuart] Remove this IP since it is no longer maintained

### LINKED ISSUE #12141: [usbuart] State of usbuart
There are several lint errors in `usbuart` due to recent port changes in `usb_fs_nb_pe`.
Is `usbuart` still maintained, and can these breakages be fixed easily?
If not we will have to remove this from lint checks when we make the AscentLint checks blocking.

### PR DESCRIPTION
**Pull request by msfschaffner**
Fixes #12141

Signed-off-by: Michael Schaffner <msf@google.com>

### CODE REVIEW

File: hw/ip/usbdev/usbdev.core [hunk 1]
```diff
@@ -3,7 +3,7 @@ CAPI=2:
 # Licensed under the Apache License, Version 2.0, see LICENSE for details.
 # SPDX-License-Identifier: Apache-2.0
 name: "lowrisc:ip:usbdev:0.1"
-description: "usbuart"
+description: "USB Device"
```
Comment (a-will) [inline hw/ip/usbdev/usbdev.core:6]: Haha, so those are the origins of this device! ;)

**Review by a-will — APPROVED** [review]

**Comment by a-will** [comment]
> ah i actually was thinking we move this out of lint but not necessarily remove it from the repo. Couldn't we leave this at D0 or something to indicate it needs to be brought up for prime time?

I'd wager usbuart will never be chosen over usbdev for an ASIC. The differences in resource utilization are likely not terribly impactful, and the NREs seem high to then limit yourself to only acting like a UART. Perhaps I'm misjudging the impact to flash and SRAM to do the UART in software atop usbdev, though.

I'm thinking usbuart is probably more oriented towards less dense / more constrained FPGA targets. However, that probably means it's not exactly an opentitan IP. Should we have an IP-specific repo where this could land? Or is it likely to simply be unmaintained regardless?

In any case, I think either path forward is fine. But even if we remove it altogether in this PR, it can always be restored from the git history, should we decide there is a use case after all.

**Comment by tjaychen** [comment]
that sounds good to me. I talked to Alex a bit and he explained that usbuart still needs software stimulus (i had thought it was kind of like an automatic block).  Given that both need to be actuated by software, i do tend to agree that usbdev would probably be the way to go.

**Review by tjaychen — APPROVED** [review]

File: hw/ip/usbdev/usbdev.core (hunk 1 above)
Comment (msfschaffner) [reply hw/ip/usbdev/usbdev.core:6]: Apparently so :)...

**Comment by msfschaffner** [comment]
Yeah I agree with that assessment. And if need be, the IP can always be restored from the Git history.

**Pull request merged**